Mathlib/Data/Set/Quot.lean

```diff
@@ -0,0 +1,77 @@
+/-
+Copyright (c) 2025 Pim Otte. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Pim Otte
+-/
+
+import Mathlib.Data.Set.Card
+
+/-!
+# Quotients as Sets
+
+## Main definitions
+
+* `Quot.toSet` interprets an equivalence class as a set.
+* `Set.Represents` says that a set represents a set of quotients if it contains exactly one element
+  from each quotient.
+-/
+
+variable {α : Type*} {r : α → α → Prop}
+
+/-- Convert an equivalence class to a `Set`. -/
+def Quot.toSet (c : Quot r) : Set α := {v | Quot.mk r v = c}
+
+instance : SetLike (Quot r) α where
+  coe := Quot.toSet
+  coe_injective' x y h := by
+    simp only [Quot.toSet] at h
+    simpa using show x.out ∈ {v | Quot.mk r v = y} by
+      rw [← h]
+      exact x.out_eq
+
+namespace Set
+
+/-- A set represents a set of quotients if it contains exactly one element from each quotient. -/
+structure Represents (s : Set α) (C : Set (Quot r)) where
```

Comment:
Is this the same as `Set.BijOn Quot.mk s C`?

Reply:
Yep. That seems very clean. I switched to that as the definition and proved the previous components as results